packages/alfa-css/src/value/numeric/length.ts

```diff
@@ -117,35 +121,57 @@ export class Length<
     hash.writeUnknown(this._math);
   }
 
-  public toJSON(): Length.JSON<CALC, U> {
-    return {
-      ...super.toJSON(),
-      value: this._value ?? null,
-      unit: this._unit ?? null,
-      math: this._math?.toJSON() ?? null,
-    };
+  public toJSON(
+    this: Length.Mixed<U>
+  ): Length.Fixed.JSON<U> | Length.Calculated.JSON {
```

Comment:
**question:** Same as other comment, can `Length.Fixed.JSON` be made non-generic?